data/basics/stateProofParticipant.go

```diff
@@ -58,13 +58,17 @@ func (p Participant) ToBeHashed() (protocol.HashID, []byte) {
 	weightAsBytes := make([]byte, 8)
 	binary.LittleEndian.PutUint64(weightAsBytes, p.Weight)
 
-	publicKeyBytes := p.PK
+	keyLifetimeBytes := make([]byte, 8)
```

Comment:
We spent some time refactoring this but ended up with lots of index calculations and had to give up this part of the code:
```
	partCommitment = append(partCommitment, weightAsBytes...)
	partCommitment = append(partCommitment, keyLifetimeBytes...)
	partCommitment = append(partCommitment, publicKeyBytes[:]...)
```
I would really like to keep this part since, in my perspective, it is readable and one can easily understand the order of the fields.
Do you have any suggestions on how to avoid the use of `make` and also use this `append` pattern?

Reply:
I don't think this is the biggest deal in the world.

Reply:
?
```
    var weightAsBytes [8]byte
    binary.LittleEndian.PutUint64(weightAsBytes[:], p.Weight)

    var keyLifetimeBytes [8]byte
    binary.LittleEndian.PutUint64(keyLifetimeBytes[:], p.PK.KeyLifetime)

    publicKeyBytes := p.PK.Commitment

    partCommitment := make([]byte, 0, len(weightAsBytes)+len(publicKeyBytes)+len(keyLifetimeBytes))
    partCommitment = append(partCommitment, weightAsBytes[:]...)
    partCommitment = append(partCommitment, keyLifetimeBytes[:]...)
    partCommitment = append(partCommitment, publicKeyBytes[:]...)
```

Reply:
Yeah, using arrays would ensure you're not allocating, just doing the double copy.